streams/src/main/java/org/apache/kafka/streams/kstream/internals/JoinedInternal.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.kstream.Joined;
+
+public class JoinedInternal<K, V, VO> extends Joined<K, V, VO>  {
+
+    JoinedInternal(final Joined<K, V, VO> joined) {
+        super(joined);
+    }
+
+    public Serde<K> keySerde() {
+        return keySerde;
+    }
+
+    public Serde<V> valueSerde() {
+        return valueSerde;
+    }
+
+    public Serde<VO> otherValueSerde() {
+        return otherValueSerde;
+    }
+
+    /**
+     * TODO rename this method to name when {@link Joined#name} will be definitely removed
+     */
+    public String internalName() {
```

Comment:
Was the intention here to avoid the deprecation warning?

If so, you can just call this method `name()` and do this:
```java
    @Override
    @SuppressWarnings("deprecation") // TODO remove this when {@link Joined#name} is removed
    public String name() {
        return name;
    }
```

Callers won't see the deprecation warning as long as they access the method via a `JoinedInternal` and not a `Joined`.

Reply:
Note: this is fixed in the next PR #6410

Reply:
We should use a different comment though:
```
@Override // TODO remove annotation when super.name() is removed
@SuppressWarnings("deprecation") // this method should not be removed if super.name() is removed
```